### PR TITLE
UI Input,  Password with Revelation

### PR DESCRIPTION
**Pull request by nhaagen**
Especially on mobile devices, I like to reveal the entered chars; this PR introduces the eye-closed/open glyphs and adds the revelation-option to password inputs.

secret
revealed

Here are the rules:
https://github.com/ILIAS-eLearning/ILIAS/pull/1119/files#diff-636dc2f66a0722b4e790c5654824ddc1

https://github.com/ILIAS-eLearning/ILIAS/pull/1119/files#diff-2d8571e3a451f51b51f9868081f6f7e8

### CODE REVIEW

**Comment by mjansenDatabay** [comment]
I like!

As far as I know @bheyser plans to use this type of field for the process described here: https://ilias.de/mantis/view.php?id=23242

File: src/UI/Component/Glyph/Factory.php [hunk 1]
```diff
@@ -675,6 +675,57 @@ public function sad($action = null);
 	 */
 	public function angry($action = null);
 
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *       The EyeClosed Glyph is used to toggle the revelation-mode of password fields.
```
Comment (Amstutz) [inline src/UI/Component/Glyph/Factory.php]: "Eye Closed" in clear text.

Comment (Amstutz) [reply src/UI/Component/Glyph/Factory.php]: Which mode is Eye Close? Letters are shown, clearly state so (effect further down indicates this I know, but make this more to the point here).

File: src/UI/Component/Glyph/Factory.php [hunk 2]
```diff
@@ -675,6 +675,57 @@ public function sad($action = null);
...
+	 *
+	 * context: Used with password-fields to toggle mask/revealed mode.
+	 *
+	 * rules:
+	 *   composition:
+	 *       1: >
+	 *          The EyeClosed Glyph MUST only be used with Password-Inputs.
+	 *   accessibility:
+	 *       1: >
+	 *          The aria-label MUST be 'eye closed'.
```
Comment (Amstutz) [inline src/UI/Component/Glyph/Factory.php]: There should be some indication about the state by the aria attributes. I would prefer something like "Eye closed - charactors of inputs fields visible".

File: src/UI/Component/Glyph/Factory.php [hunk 3]
```diff
@@ -675,6 +675,57 @@ public function sad($action = null);
...
+	 * @param	string|null	$action
+	 * @return	\ILIAS\UI\Component\Glyph\Glyph
+	 */
+	public function eyeclosed($action = null);
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *       The EyeOpen Glyph is used to toggle the revelation-mode of password fields.
```
Comment (Amstutz) [inline src/UI/Component/Glyph/Factory.php]: Eye Open

Comment (Amstutz) [reply src/UI/Component/Glyph/Factory.php]: Same as above, leave some state indication.

File: src/UI/Component/Glyph/Factory.php [hunk 4]
```diff
@@ -675,6 +675,57 @@ public function sad($action = null);
...
+	 *
+	 * context: Used with password-fields to toggle mask/revealed mode.
+	 *
+	 * rules:
+	 *   composition:
+	 *       1: >
+	 *          The EyeOpen Glyph MUST only be used with Password-Inputs.
+	 *   accessibility:
+	 *       1: >
+	 *          The aria-label MUST be 'eye open'.
```
Comment (Amstutz) [inline src/UI/Component/Glyph/Factory.php]: Same as above, make state clear here for blind people.

**Review by Amstutz — CHANGES_REQUESTED** [review]
LGTM. Some minor remarks about description.

However, there is one major point:
- I believe the aria label here is crucial for blind people. Imagine somebody without sight sitting in a public place revealing the password to other people without knowing, since the label for revealing or not was not clear. Would be bad. Improve the aria label.

**Comment by matthiaskunkel** [comment]
Jour Fixe, 30 JUL 2018 : We highly appreciate this PR. Please consider Timon's comments and offer a solution before committing it to trunk.

**Comment by klees** [comment]
@Amstutz May we merge this?

**Pull request merged**